processor/tailsamplingprocessor/metrics.go

```diff
@@ -43,6 +43,11 @@ var (
 	statTracesOnMemoryGauge     = stats.Int64("sampling_traces_on_memory", "Tracks the number of traces current on memory", stats.UnitDimensionless)
 )
 
+func init() {
```

Comment:
Might be better to register them once the processor is created, like here:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e1a9ff10219bdf53cd51eb35fa1ba6268114037d/processor/groupbytraceprocessor/factory.go#L45-L53

Otherwise, you'll register the views even when this processor isn't being used.